unshieldv3: init at 0.2.2

jchv opened the pull request:
Adds `unshieldv3`, a tool to extract InstallShield V3 .Z archive files. These files are not handled by `unshield`, which handles .CAB archives from later InstallShield versions.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

jchv committed:
unshieldv3: init at 0.2.2

jchv commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 401312`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>unshieldv3</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>unshieldv3</li>
  </ul>
</details>

phanirithvij commented:
# [nixpkgs-reviewd](https://github.com/nixosbrasil/nixpkgs-reviewd) workflow finished

**Status and logs**: https://github.com/phanirithvij/nixpkgs-reviewd/actions/runs/14633571868


## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 401312`

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>unshieldv3</li>
  </ul>
</details>

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 401312`

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>unshieldv3</li>
  </ul>
</details>

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 401312`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>unshieldv3</li>
  </ul>
</details>

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 401312`

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>unshieldv3</li>
  </ul>
</details>

illdefined commented on pkgs/by-name/un/unshieldv3/package.nix:
```diff
@@ -0,0 +1,52 @@
+
+  meta = {
+    description = "Tool to extract .Z files from InstallShield V3 (Z) installers";
+    homepage = "https://github.com/wfr/unshieldv3";
```
```suggestion
    homepage = "https://github.com/wfr/unshieldv3";
    changelog = "https://github.com/wfr/unshieldv3/releases/tag/v${finalAttrs.version}"
```

jchv replied:
Added.

illdefined commented on pkgs/by-name/un/unshieldv3/package.nix:
```diff
@@ -0,0 +1,52 @@
+  src = fetchFromGitHub {
+    owner = "wfr";
+    repo = "unshieldv3";
+    rev = "v${finalAttrs.version}";
```
```suggestion
    tag = "v${finalAttrs.version}";
```

jchv replied:
Good call, done.

illdefined commented on pkgs/by-name/un/unshieldv3/package.nix:
```diff
@@ -0,0 +1,52 @@
+  };
+
+  patches = [
+    (fetchpatch {
+      url = "https://github.com/wfr/unshieldv3/commit/99a5cf5e04e67f37a58620343079b6c6a97e4eb3.patch";
+      hash = "sha256-yElDUxAchMLY784u4i7++YQvUlMGyDfhwCAi6PzExrI=";
+    })
```
```suggestion
    # enable build on Clang
    (fetchpatch {
      url = "https://github.com/wfr/unshieldv3/commit/99a5cf5e04e67f37a58620343079b6c6a97e4eb3.patch";
      hash = "sha256-yElDUxAchMLY784u4i7++YQvUlMGyDfhwCAi6PzExrI=";
    })
```

jchv replied:
Oops, yep, that's a good idea. Added a comment.

illdefined commented on pkgs/by-name/un/unshieldv3/package.nix:
```diff
@@ -0,0 +1,52 @@
+  nativeBuildInputs = [ cmake ];
+
+  doInstallCheck = true;
+  installCheckPhase = ''
+    runHook preInstallCheck
+
+    for i in $src/test-data/*.Z; do
+      mkdir -p test
+      $out/bin/unshieldv3 extract $i test
+    done
+
+    runHook postInstallCheck
+  '';
```
Maybe that is being pedantic, do you think that this could be done in `checkPhase` or `postCheck`?

jchv replied:
Sure. I went with `postCheck` to avoid overriding the `checkPhase`, in case the upstream ever adds CMake tests.

illdefined reviewed (changes requested):
I checked compilation on `x86_64-linux` and `aarch64-linux`. The package outputs look clean and the produced binaries can be run on both platforms.

I verified that the licence information is consistent with the licence in the upstream repo.

The package itself looks very good and I only have minor nitpicks or suggestions.

illdefined commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 401312`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>unshieldv3</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>unshieldv3</li>
  </ul>
</details>

illdefined commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 401312`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>unshieldv3</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>unshieldv3</li>
  </ul>
</details>

illdefined reviewed (approved):
Thank you!

illdefined reviewed (approved):
Just confirming that the last change has no impact on package outputs. The patches are identical except for being based on different commit IDs.

jchv commented:
Updated to v0.2.2 which integrates the patch. Dropped the patch. Re-tested the execution, appears to be good.

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 401312`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>unshieldv3</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>unshieldv3</li>
  </ul>
</details>

illdefined commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 401312`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>unshieldv3</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>unshieldv3</li>
  </ul>
</details>
